usaspending_api/references/v2/views/toptier_agencies.py

```diff
@@ -59,73 +50,148 @@ def get(self, request, format=None):
-                treasury_account_identifier__funding_toptier_agency=toptier_agency,
-            ).aggregate(
+            .order_by("toptier_code", "-reporting_fiscal_year", "-reporting_fiscal_quarter", "-reporting_fiscal_period")
+            .distinct("toptier_code")
+            .values("toptier_code", "reporting_fiscal_year", "reporting_fiscal_quarter", "reporting_fiscal_period")
+        )
+        latest_submission_by_toptier = {
+            val["toptier_code"]: {
+                "fiscal_year": val["reporting_fiscal_year"],
+                "fiscal_quarter": val["reporting_fiscal_quarter"],
+                "fiscal_period": val["reporting_fiscal_period"],
+            }
+            for val in latest_submission_by_toptier
+        }
+
+        # Add a placeholder for Agencies that have submitted but not in recent Quarter / Period;
+        # Placeholder is most recent Period window to ensure accurate "current_total_budget_authority_amount"
+        placeholder = {
+            "fiscal_year": most_recent_period_window["submission_fiscal_year"],
+            "fiscal_quarter": most_recent_period_window["submission_fiscal_quarter"],
+            "fiscal_period": most_recent_period_window["submission_fiscal_month"],
+        }
+        for toptier_code in all_toptier_with_submission:
+            submission_values = latest_submission_by_toptier.get(toptier_code)
+            if submission_values is None:
+                latest_submission_by_toptier[toptier_code] = placeholder
+
+        aab_sums_by_toptier = (
+            AppropriationAccountBalances.objects.filter(submission__is_final_balances_for_fy=True)
+            .values(
+                "treasury_account_identifier__funding_toptier_agency",
+                "submission__reporting_fiscal_year",
+                "submission__reporting_fiscal_quarter",
+            )
+            .annotate(
                 budget_authority_amount=Coalesce(Sum("total_budgetary_resources_amount_cpe"), 0),
                 obligated_amount=Coalesce(Sum("obligations_incurred_total_by_tas_cpe"), 0),
                 outlay_amount=Coalesce(Sum("gross_outlay_amount_by_tas_cpe"), 0),
             )
+        )
+        aab_sums_by_toptier = {
+            (
+                val["treasury_account_identifier__funding_toptier_agency"],
+                val["submission__reporting_fiscal_year"],
+                val["submission__reporting_fiscal_quarter"],
+            ): {
+                "budget_authority_amount": val["budget_authority_amount"],
+                "obligated_amount": val["obligated_amount"],
+                "outlay_amount": val["outlay_amount"],
+            }
+            for val in aab_sums_by_toptier
+        }
 
-            abbreviation = ""
-            if toptier_agency.abbreviation is not None:
-                abbreviation = toptier_agency.abbreviation
+        # get agency queryset, distinct toptier id to avoid duplicates, take first ordered agency id for consistency
```

Comment:
The sort on `agency.id` was most likely put in place for the purpose of the `distinct` since Django documentation recommends having a sort to guarantee the result (instead of letting Django do what it wants). Since the values being used from this are all from the Toptier Agency the `agency.id` it does not affect the the end result at all. 

Would recommend not changing this though since the Agency ID does match what the Agency endpoint is using to determine which Agency you are looking at.